runner.go

```diff
@@ -270,6 +248,32 @@ func (runner *Runner) StopWorker(id string) error {
 	return ErrDead
 }
 
+// StopAndRemoveWorker stops the worker and returns any error reported by
+// the worker, waiting for the worker to be no longer known to the runner.
+// If it was stopped while waiting, StopAndRemoveWorker will return ErrAborted.
+//
+// StopAndRemoveWorker returns ErrDead if the runner is not running.
+func (runner *Runner) StopAndRemoveWorker(id string, abort <-chan struct{}) error {
+	// workerInfo contains the done channel to signal that the worker has
+	// been removed from the runner.
+	// If it's not there, then Worker() returns not found anyway.
+	runner.mu.Lock()
+	workerInfo, ok := runner.workers[id]
+	runner.mu.Unlock()
+	w, err := runner.Worker(id, abort)
```

Comment:
this is superfluous, runner.Worker might wait for the worker but if it didn't exist in the `runner.workers[id]` its going to exit below with a nil error

Reply:
yeah but there is a race condition between `runner.workers[id]` and `runner.Worker(id, abort)`. If the workerInfo didn't exist but then it did the if statement below would fail, despite the workerinfo and worker now existing